plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/edit/SQLServerIndexManager.java

```diff
@@ -78,7 +77,9 @@ protected void addObjectCreateActions(@NotNull DBRProgressMonitor monitor, @NotN
         }
         if (index.isPersisted()) {
             try {
-                String indexDDL = index.getObjectDefinitionText(monitor, DBPScriptObject.EMPTY_OPTIONS);
+
+                options.put("isUnique", command.getObject().isUnique());
```

Comment:
Looks like over-engineering.
Why use so complicated logic, why index itself cannot reqlize that it is unique (in getObjectDefinitionText)?